### PR TITLE
Use the spread LXD backend in Travis

### PR DESCRIPTION
**Pull request by RAOF**

### CODE REVIEW

**Comment by RAOF** [comment]
Note that this *currently* pulls a working spread from my own server. Apparently I don't have an account on people.canonical.com?

File: spread/build/fedora/task.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+summary: Build (on Ubuntu)
```
Comment (Saviq) [inline spread/build/fedora/task.yaml]: on Fedora, no?

File: .travis.yml [hunk 2]
```diff
@@ -10,19 +10,26 @@ branches:
...
+  - sudo su travis -c 'lxc network attach-profile lxdbr0 default eth0';
+  - wget https://cooperteam.net/spread_2017.09.21_amd64.snap
+  - sudo snap install --dangerous spread_2017.09.21_amd64.snap
+
 env:
-  global:
-    # SPREAD_LINODE_KEY
-  - secure: KT6a0jTsWX1jQq9vlG1F6dE7Orn/VbiIQihe+v/VBYXiAvAKaoK520MUBgdaf3svULRwyn1T6yKOc3IcAs+L5qA6/oeyeLF+xC0ckpNaTEHSov5YJ+hNlL8bL1rZ1U9bZBXMAuhkjHuRi3agk+HgD1CI65xOiO7txyGArEF1qOlWR2dskWYKh+EuNEohl28OIi6iFZrMfCFrzG8sZ6V0RuS1nakhitnsJ87aZ1LNedDY4diL/DdrmSg4zgC81dH+etDlBDOf1amkj6/s/IgUAgeLmyuENjXXd1LQVjL6MjzLpot4BlcYoD+EHpvH9kbUe69t5LW4bmmJdRBa37l6T25CYrX/9oG83WDGr/FXCxUGEXhplaSkoGmga/PB+Zxya91d850lu+r2aOz/6FwHnwYseVxQef/hemE40QvuDDTX2tjKMoiw3rLktuAJTJfiSE84bjSPCM49aUemLpqTYk2w/avPWvv7zVtpgFDR9dkjti9p+FPKaqIuZ+XQFJ9Ov+MmCkOuBA1ieZN5I2+Exr6UKDbIBz8h97cqag7mcUB5QlMU/Iu7KiMr7NdHFBny3DHVgQtxAcC5yzuzPpa0zkJu9pnaCKy7RC+/uS/zCY05zGhyC8Qfdg3H0da9NlKpLUU6ojjPkuaZMpKeGkyMcYaMVjoaxdT2lRgtQFF9W1s=
+        #  global:
+        #   - LXD_DIR=/var/snap/lxd/common/lxd
```
Comment (Saviq) [inline .travis.yml]: Want to leave these be?

File: .travis.yml [hunk 3]
```diff
@@ -10,19 +10,26 @@ branches:
   - trying
   - master
 
+install:
+  - sudo add-apt-repository -y ppa:ubuntu-lxc/lxd-stable;
+  - sudo apt-get -qq update;
+  - sudo apt-get -y install lxd;
```
Comment (Saviq) [inline .travis.yml]: Could use the LXD snap instead?

File: .travis.yml [hunk 4]
```diff
@@ -10,19 +10,26 @@ branches:
...
-  - SYSTEM=ubuntu-17.10 VARIANT=gcc
-  - SYSTEM=ubuntu-17.10 VARIANT=clang
-
-before_install:
-  - sudo apt-get --yes install snapd
-  - sudo snap install spread
+   - SYSTEM=ubuntu-16.04 VARIANT=gcc
+   - SYSTEM=ubuntu-17.10 VARIANT=gcc
+   - SYSTEM=ubuntu-17.10 VARIANT=clang
+   - SYSTEM=fedora-26 VARIANT=gcc
```
Comment (Saviq) [inline .travis.yml]: Sure you want these whitespace changes?

File: .travis.yml [hunk 5]
```diff
@@ -10,19 +10,26 @@ branches:
...
 
 script:
-  - /snap/bin/spread linode:$SYSTEM:...:$VARIANT
+  - sudo su $USER -c '/snap/bin/spread lxd:$SYSTEM:...:$VARIANT'
```
Comment (Saviq) [inline .travis.yml]: This is to get the new group? Would `sg` work?

File: .travis.yml (hunk 3 above)
Comment (RAOF) [reply .travis.yml]: I tried that first (as the commented-out LXD_DIR environment might suggest 😄); The LXD snap doesn't seem to work in Travis.

File: spread/build/fedora/task.yaml (hunk 1 above)
Comment (RAOF) [reply spread/build/fedora/task.yaml]: Ahem, yes.

File: .travis.yml (hunk 5 above)
Comment (RAOF) [reply .travis.yml]: Turns out, yes!

**Comment by RAOF** [comment]
I've switched to using `sg`, which does indeed work.

I've removed the commented out bits entirely; they can be restored from git if we need!

**Comment by Saviq** [comment]
Looks good

bors r+

**Pull request merged**